### PR TITLE
Fix for the water use inconsistent corner case

### LINKED ISSUE #9777: A possibly inconsistent corner case for WaterUse:Equipment when target water temperature schedule is left blank
Issue overview
--------------
It was reported by a user that there could be some inconsistency between v22.1 and v22.2 on the treatment a blank “Target temperature” input field in WaterUse:Equipment. In the old v22.1 EnergyPlus program, (and also stated so in the v22.2 IORef documentation), the default treatment for a blank “target temperature schedule” is to use the hot water schedule instead. However, in the actual v22.2 EnergyPlus program, this is not the case---the user’s observation via a few experiments showed that the related “Water Use Equipment Heating Energy” will be zero---which seems unreasonable if the blank “target temperature schedule” is indeed defaults to the “hot water schedule” temperature. While in previous version (e.g. v22.1) this Energy value is non-zero, which sounds a more consistent reasonable with the stated assumption about the default treatment (of using “hot water scheduled temperature” when the “target temperature schedule” is left blank).

After some digging in the code (and also reading some of the comment lines), it was found that in the v22.2 code, the default treatment could be different: the “target water schedule temperature” if left blank, will now actually default to the “cold water scheduled” temperature instead (different from what is stated in the v22.2 IORef): 
https://github.com/NREL/EnergyPlus/blob/2d609ecfefd7afe1eb4e8b416f16fe4fc1e1e8e7/src/EnergyPlus/WaterUse.cc#L587-L591

However, in the code, there are also places that still trying to stick to the older convention (using hotwater temperature, for example here: 
https://github.com/NREL/EnergyPlus/blob/2d609ecfefd7afe1eb4e8b416f16fe4fc1e1e8e7/src/EnergyPlus/WaterUse.cc#L1055-L1057
Actually in order for this logic here to work out:
https://github.com/NREL/EnergyPlus/blob/2d609ecfefd7afe1eb4e8b416f16fe4fc1e1e8e7/src/EnergyPlus/WaterUse.cc#L1211-L1215
The place in L1055-L1057 above should use cold water temperature instead for the blank target schedule scenario.

This might need to be addressed, one way or the other---either (Option 1) make it all consistent with older version treatment; or (Option 2) make some revisions in a couple of places in the code and the IORef document to clearly support/state the new way of treating this.

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version): All
 - Version of EnergyPlus (if using an intermediate build, include SHA): v22.2
 - Related helpdesk ticket number: 16314

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jcyuan2020**
Pull request overview
---------------------
 - Fixes #9777, using the proposed "Option 2" described in the issue post.
 - A needed change is to set the default target water temperature to either the cold water temperature or the hot water temperature depending on different scenarios: if no water connection is used, and if the hot water schedule is not set, the target temperature will default to the cold water supply temperature; if there is water connection used for the water use equipment, or there is a hot water schedule present, the target water temperature will default to the hot water supply temperature.
 - The input field descriptions in the IORef is also modified to reflect the logic change based on the scenarios described above.

- There could be an IDD change about the note in this field. But at this point, this note changed will have to wait till after the official release.

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/WaterUse.cc [hunk 1]
```diff
@@ -1052,8 +1052,8 @@ namespace WaterUse {
 
         if (this->TargetTempSchedule > 0) {
             this->TargetTemp = ScheduleManager::GetCurrentScheduleValue(state, this->TargetTempSchedule);
-        } else { // If no TargetTempSchedule, use all hot water
-            this->TargetTemp = this->HotTemp;
+        } else { // If no TargetTempSchedule, use all cold water
+            this->TargetTemp = this->ColdTemp;
```
Comment (rraustad) [inline src/EnergyPlus/WaterUse.cc:1059]: OK, so there is a TargetTempSchedule (mixed water temp), a ColdTempSchedule, and a HotTempSchedule. The Cold/HotTempSched's should have no impact on mixed water temp. So if no TargetTempSchedule exists then it does make sense to use the ColdTemp.

However, this logic suggests that there are times when the HotTemp should be used.

    waterEquipment.allowHotControl = (waterEquipment.TargetTempSchedule && waterEquipment.HotTempSchedule) || waterEquipment.Connections; 

Maybe this logic should include the allowHotControl variable to cover the case for waterEquipment.Connections?

        if (this->TargetTempSchedule > 0) {
            this->TargetTemp = ScheduleManager::GetCurrentScheduleValue(state, this->TargetTempSchedule);
        } else if (waterEquipment.allowHotControl )  { // If no TargetTempSchedule, use all hot water if applicable
            this->TargetTemp = this->HotTemp;
        } else { // If no TargetTempSchedule, use all cold water
            this->TargetTemp = this->ColdTemp;
        }

Comment (rraustad) [reply src/EnergyPlus/WaterUse.cc:1059]: Interesting notes in the idd:

    WaterUse:Equipment,
       \memo A generalized object for simulating all water end uses. Hot and cold water uses are
       \memo included, as well as controlled mixing of hot and cold water at the tap. The
       \memo WaterUse:Equipment object can be used stand-alone, or coupled into a plant loop using
       \memo the WaterUse:Connections object (see below). The WaterUse:Connections object allows
       \memo water uses to be linked to WaterUse:Storage objects to store and draw reclaimed water.
       \memo The object can also simulate drainwater heat recovery.
    A4 , \field Target Temperature Schedule Name
       \type object-list
       \object-list ScheduleNames
       \note Defaults to hot water supply temperature
    A5 , \field Hot Water Supply Temperature Schedule Name
       \type object-list
       \object-list ScheduleNames
       \note Defaults to cold water supply temperature
    A6 , \field Cold Water Supply Temperature Schedule Name
       \type object-list
       \object-list ScheduleNames
       \note Defaults to water temperatures calculated by Site:WaterMainsTemperature object

Comment (jcyuan2020) [reply src/EnergyPlus/WaterUse.cc:1059]: So here it looks like that (first?) we will need to determine (according to idd, code) what the default target water temperature should go to for the "target temperature schedule = blank case": if determine that this should be default to cold water, then the IDD description needs to be changed here (as well as the IORef doc descriptions, which is already changed for the "Option 2" fix). In this case, if the 22.2 new feature/fix ( mixing cold/hot relationship) decided this should be the case, then we will change the IDD description as well.

If we determined that here that the target water temperature should take the hot water temperature, then we should go to the "Option 1" way of fix, to find ways to make the code to use hot water temperature as the default target temperature when the target temperature schedule field is blank. The good side is that it seems to be consistent to what the older versions (<=v22.1) did; and no IORef or IDD change is required.

Comment (jcyuan2020) [reply src/EnergyPlus/WaterUse.cc:1059]: > 
Trying to figure out the proposed adding the logic of  
```
 } else if (waterEquipment.allowHotControl )  { // If no TargetTempSchedule, use all hot water if applicable
        this->TargetTemp = this->HotTemp;
```

1) Here the block is already the situation where target water temperature schedule is not present (0); then autoHotControl will be true only if there is water connection. So this logic basically will be true and executed only if there is water connection. If this is the case, and in this case the hot water temperature should be the default target temperature, then it is the right logic.

2) Then the next block of setting cold water temperature as the default target water temperature will only be executed when there is be no water connection (and no target water schedule temperature field, as it is also in the situation of target water temperature schedule is not present). 

So we will need to decide the logics above would be the correct  (or expected) logics.

**Comment by rraustad** [comment]
I would suggest that the defect file for #9651 and example files that had diffs be tested and plotted to be sure results have not changed (or be explained) from what was found when this issue was corrected in #9661. One line changes always have more work than first thought.

**Comment by nrel-bot-3** [comment]
@jcyuan2020 @myoldmopar it has been 28 days since this pull request was last updated.

File: src/EnergyPlus/WaterUse.cc (hunk 1 above)
Comment (rraustad) [reply src/EnergyPlus/WaterUse.cc:1059]: I do like this suggested logic:

    if (this->TargetTempSchedule > 0) {
        this->TargetTemp = ScheduleManager::GetCurrentScheduleValue(state, this->TargetTempSchedule);
    } else if (waterEquipment.allowHotControl )  { // If no TargetTempSchedule, use all hot water if applicable
        this->TargetTemp = this->HotTemp;
    } else { // If no TargetTempSchedule, use all cold water
        this->TargetTemp = this->ColdTemp;
    }

except that it relies on this logic being accurate:

    waterEquipment.allowHotControl = (waterEquipment.TargetTempSchedule && waterEquipment.HotTempSchedule) || waterEquipment.Connections;

Is it possible the user had other intentions with either of these inputs? Is there another corner case we are missing? If there is any doubt about the above logic then the user would have to provide an input to specify the control. Maybe the suggestion at the top of this comment is sufficient? and we wait for the next corner case to pop up. Does this suggestion fix the defect file?

**Comment by Myoldmopar** [comment]
@jcyuan2020 this is currently marked as draft. That is totally fine, but if you plan on getting it in for 23.1, please add that milestone, or comment back here and I can do it.

**Comment by jcyuan2020** [comment]
Added to the milestone.

**Comment by Myoldmopar** [comment]
One more quick thing, @jcyuan2020, the title still says "Tentative Fix".  Could you update the title accordingly?

**Comment by jcyuan2020** [comment]
The PR title is now updated.

**Comment by Myoldmopar** [comment]
Thanks.  There is a [build warning ](https://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-9e05ec8390db9f7083e53e8462d559d90322792d-PR9778-x86_64-MacOS-10.17-clang-13.0.0.html) that I'll fix real quick and run a test.  Anticipate this to merge shortly after I push the warning fix.

**Comment by Myoldmopar** [comment]
OK, this is good to go now.  Thanks @jcyuan2020, merging.

**Pull request merged**